native-audio/app/src/main/java/com/example/nativeaudio/NativeAudio.java

```diff
@@ -155,12 +170,19 @@ public void onClick(View view) {
 
         ((Button) findViewById(R.id.pause_uri)).setOnClickListener(new OnClickListener() {
             public void onClick(View view) {
+                if(!permissionGranted) {
+                    // Log.e(TAG, "Error: requested permission not granted, please restart app and grant access");
```

Comment:
not sure about this one: to follow the tradition for this sample, I should not add any log: original author removed all of the logs before he checked in code [ he left his TAG there commented out for later use I guess ]. So I followed his tradition to leave some hint there but not enabled it.